cms/djangoapps/maintenance/views.py

```diff
@@ -211,3 +227,77 @@ def post(self, request):
             exc_info=True
         )
         return self.render_response()
+
+
+class AnnouncementBaseView(View):
+    """
+    Base view for Announcements pages
+    """
+
+    @method_decorator(require_global_staff)
+    def dispatch(self, request, *args, **kwargs):
+        return super(AnnouncementBaseView, self).dispatch(request, *args, **kwargs)
+
+
+class AnnouncementIndexView(ListView, MaintenanceBaseView):
+    """
+    View for viewing the announcements shown on the dashboard, used by the global staff.
+    """
+    model = Announcement
+    object_list = Announcement.objects.all()
+    context_object_name = 'announcement_list'
+    paginate_by = 9
+
+    def __init__(self):
+        super(AnnouncementIndexView, self).__init__(MAINTENANCE_VIEWS['announcement_index'])
+
+    def get_context_data(self, **kwargs):
+        context = super(AnnouncementIndexView, self).get_context_data(**kwargs)
+        context['view'] = MAINTENANCE_VIEWS['announcement_index']
+        return context
+
+    @method_decorator(require_global_staff)
+    def get(self, request, *args, **kwargs):
+        if "limit" in request.GET.keys():
+            self.paginate_by = max(request.GET.get("limit"), 50)
+        context = self.get_context_data()
+        return render_to_response(self.template, context)
+
+
+class AnnouncementEditView(UpdateView, AnnouncementBaseView):
+    """
+    View for editing an announcement.
+    """
+    model = Announcement
+    form_class = AnnouncementForm
+    success_url = '/maintenance/announcements'
```

Comment:
Should use `reverse_lazy('maintenance:announcement_index')` instead of hardcoded path (cf [django docs for why `reverse_lazy`](https://docs.djangoproject.com/en/1.11/topics/class-based-views/generic-editing/#model-forms)).